exchanges/lbank/lbank_wrapper.go

```diff
@@ -0,0 +1,523 @@
+	var finalResp []exchange.OrderDetail
+	var resp exchange.OrderDetail
+	for a := range getOrdersRequest.Currencies {
+		p := exchange.FormatExchangeCurrency(l.Name, getOrdersRequest.Currencies[a])
+		b := int64(1)
+		tempResp, err := l.QueryOrderHistory(p.String(), strconv.FormatInt(b, 10), "200")
+		if err != nil {
+			return finalResp, err
+		}
+		for tempResp.PageLength == 200 {
+			tempResp, err = l.QueryOrderHistory(p.String(), strconv.FormatInt(b, 10), "200")
+			if err != nil {
+				return finalResp, err
+			}
+			for x := 0; x < len(tempResp.Orders); x++ {
+				resp.Exchange = l.GetName()
+				resp.CurrencyPair = currency.NewPairFromString(tempResp.Orders[x].Symbol)
+				if strings.EqualFold(tempResp.Orders[x].Type, "buy") {
+					resp.OrderSide = exchange.BuyOrderSide
+				} else {
+					resp.OrderSide = exchange.SellOrderSide
+				}
+				z := tempResp.Orders[x].Status
+				switch {
+				case z == -1:
+					resp.Status = "cancelled"
+				case z == 1:
+					resp.Status = "on trading"
+				case z == 2:
+					resp.Status = "filled partially"
+				case z == 3:
+					resp.Status = "Filled totally"
+				case z == 4:
+					resp.Status = "Cancelling"
+				default:
+					return finalResp, fmt.Errorf("invalid order status: %v", tempResp.Orders[x].Status)
+				}
+				resp.Price = tempResp.Orders[x].Price
+				resp.Amount = tempResp.Orders[x].Amount
+				resp.OrderDate = time.Unix(tempResp.Orders[x].CreateTime, 9)
+				resp.ExecutedAmount = tempResp.Orders[x].DealAmount
+				resp.RemainingAmount = tempResp.Orders[x].Price - tempResp.Orders[x].DealAmount
+				resp.Fee, err = l.GetFeeByType(&exchange.FeeBuilder{
+					FeeType:       exchange.CryptocurrencyTradeFee,
+					Amount:        tempResp.Orders[x].Amount,
+					PurchasePrice: tempResp.Orders[x].Price})
+				if err != nil {
+					return finalResp, err
+				}
+				finalResp = append(finalResp, resp)
+				b++
+			}
+		}
+	}
+	return finalResp, nil
+}
+
+// GetFeeByType returns an estimate of fee based on the type of transaction *
+func (l *Lbank) GetFeeByType(feeBuilder *exchange.FeeBuilder) (float64, error) {
+	var resp float64
+	if feeBuilder.FeeType == exchange.CryptocurrencyTradeFee {
+		return feeBuilder.Amount * feeBuilder.PurchasePrice * l.Fee, nil
+	}
+	if feeBuilder.FeeType == exchange.CryptocurrencyWithdrawalFee {
+		withdrawalFee, err := l.GetWithdrawConfig(feeBuilder.Pair.Base.Lower().String())
+		if err != nil {
+			return resp, err
+		}
+		resp, err = strconv.ParseFloat(withdrawalFee[0].Fee, 64)
+		if err != nil {
+			return resp, err
+		}
+	}
+	return resp, nil
+}
+
+// GetAllOpenOrderID returns map[string][]string -> map[currencypair][]orderIDs
+func (l *Lbank) GetAllOpenOrderID() (map[string][]string, error) {
+	allPairs := l.GetEnabledCurrencies()
+	resp := make(map[string][]string)
+	for a := range allPairs {
+		p := exchange.FormatExchangeCurrency(l.Name, allPairs[a])
+		b := int64(1)
+		tempResp, err := l.GetOpenOrders(p.String(), strconv.FormatInt(b, 10), "200")
```

Comment:
Nothing is done with this response if pagelength is less than 200 and this function returns an empty map